RetailCoder.VBE/Inspections/QuickFixes/AssignedByValParameterMakeLocalCopyQuickFix.cs

```diff
@@ -0,0 +1,182 @@
+﻿using Rubberduck.Inspections.Abstract;
+using System.Linq;
+using Rubberduck.Parsing;
+using Rubberduck.VBEditor;
+using Rubberduck.Inspections.Resources;
+using Rubberduck.Parsing.Grammar;
+using Rubberduck.Parsing.Symbols;
+using System.Windows.Forms;
+using Rubberduck.UI.Refactorings;
+using Rubberduck.Common;
+using Antlr4.Runtime;
+using System.Collections.Generic;
+using Antlr4.Runtime.Tree;
+
+namespace Rubberduck.Inspections.QuickFixes
+{
+    public class AssignedByValParameterMakeLocalCopyQuickFix : QuickFixBase
+    {
+        private readonly Declaration _target;
+        private string _localCopyVariableName;
+        private string[] _variableNamesAccessibleToProcedureContext;
+        private bool _isQuickFixUnitTest;
+
+        public AssignedByValParameterMakeLocalCopyQuickFix(Declaration target, QualifiedSelection selection)
+            : base(target.Context, selection, InspectionsUI.AssignedByValParameterMakeLocalCopyQuickFix)
+        {
+            _target = target;
+            _isQuickFixUnitTest = false;
+            _localCopyVariableName = "x" + _target.IdentifierName.CapitalizeFirstLetter();
+            _variableNamesAccessibleToProcedureContext = GetVariableNamesAccessibleToProcedureContext(_target.Context.Parent.Parent);
+        }
+
+        public override bool CanFixInModule { get { return false; } }
+        public override bool CanFixInProject { get { return false; } }
+
+        //This function exists solely to support unit testing 
+        public void TESTONLY_FixUsingAutoGeneratedName(string suggestedName = "")
+        {
+            //Prevent the popup dialog and forces the use of the Suggested Name
+            _isQuickFixUnitTest = true;
+            if (suggestedName.Length > 0)
+            {
+                _localCopyVariableName = suggestedName;  //unit test poses as user input
+            }
+
+            Fix();
+        }
+
+        public override void Fix()
+        {
+            SetLocalCopyVariableName();
+
+            if (!ProposedLocalVariableNameIsValid() || IsCancelled)
+            {
+                return;
+            }
+
+            ReplaceAssignedByValParameterReferences();
+
+            InsertLocalVariableDeclarationAndAssignment();
+        }
+
+        private void SetLocalCopyVariableName()
+        {
+            using (var view = new AssignedByValParameterQuickFixDialog(_target, Selection))
+            {
+                view.NewName = _localCopyVariableName;
+                view.IdentifierNamesAlreadyDeclared = _variableNamesAccessibleToProcedureContext;
+                if (!_isQuickFixUnitTest)
```

Comment:
Thanks for providing all this detail.  I'll look at the examples and get rid of the dependency (I was a little surprised this topic didn't come up in the very first PR!).